pilot/pkg/model/network.go

```diff
@@ -79,20 +80,55 @@ func NewNetworkManager(env *Environment, xdsUpdater XDSUpdater) (*NetworkManager
 }
 
 func (mgr *NetworkManager) reloadAndPush() {
+	changed := mgr.reloadNetworkEndpoints()
```

Comment:
I think you should keep `reloadNetworkEndpoints` as another network handler and only register to `NetworksWatcher`

Reply:
SGTM. Done